### PR TITLE
DHFPROD-1932: UI Implementation of Adding and removing flow steps

### PR DESCRIPTION
**Pull request by brucean52**

### CODE REVIEW

File: quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/edit-flow-ui.component.ts [hunk 1]
```diff
@@ -58,7 +59,44 @@ export class EditFlowUiComponent {
...
+  }
+  deleteFlowDialog(): void {
+    const dialogRef = this.dialog.open(ConfirmationDialogComponent, {
+      width: '350px',
+      data: {title: 'Delete Flow?', confirmationMessage: `Delete ${this.flow.name}?`}
+    });
+
+    dialogRef.afterClosed().subscribe(response => {
+      if (response) {
+
```
Comment (alexsmr) [inline quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/edit-flow-ui.component.ts]: There is a service implementation of deleting the flow. Please use it.

File: quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/edit-flow-ui.component.ts [hunk 2]
```diff
@@ -58,7 +59,44 @@ export class EditFlowUiComponent {
...
+    });
+  }
+  redeployDialog(): void {
+    const dialogRef = this.dialog.open(ConfirmationDialogComponent, {
+      width: '350px',
+      data: {title: 'Redeploy Flow?', confirmationMessage: `Redeploy ${this.flow.name} to database?`}
+    });
 
+    dialogRef.afterClosed().subscribe(response => {
+      if (response) {
```
Comment (alexsmr) [inline quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/edit-flow-ui.component.ts]: Please add some log for this.

File: quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts [hunk 3]
```diff
@@ -8,10 +9,27 @@ import { Component, Input, Output, EventEmitter, ViewChild } from '@angular/core
 })
 export class StepComponent {
   @Input() step: any;
-  constructor(
+  @Input() databases: any;
```
Comment (alexsmr) [inline quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts]: What is "databases" refers to?

File: quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts [hunk 4]
```diff
@@ -8,10 +9,27 @@ import { Component, Input, Output, EventEmitter, ViewChild } from '@angular/core
...
+    const dialogRef = this.dialog.open(NewStepDialogComponent, {
+      width: '600px',
+      data: {databases: this.databases}
+    });
 
-
+    dialogRef.afterClosed().subscribe(response => {
+      if (response) {
+        // TODO Add Step to backend
+        // this.flow.steps.push(response);
```
Comment (alexsmr) [inline quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts]: Please ask around about the usage of service. The implementation should be down to the service level.

File: quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts (hunk 3 above)
Comment (brucean52) [reply quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts]: databases is for the dropdown of databases the user can select from. ex staging, final, and jobs

File: quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/edit-flow-ui.component.ts (hunk 1 above)
Comment (brucean52) [reply quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/edit-flow-ui.component.ts]: added the service in latest PR.

**Pull request merged**